system-reinstall-bootc/src/users.rs

```diff
@@ -102,61 +103,134 @@ impl Display for UserKeys {
     }
 }
 
+#[derive(Debug)]
+struct SshdConfig<'a> {
+    authorized_keys_files: Vec<&'a str>,
+    authorized_keys_command: &'a str,
+    authorized_keys_command_user: &'a str,
+}
+
+impl<'a> SshdConfig<'a> {
+    pub fn parse(sshd_output: &'a str) -> Result<SshdConfig<'a>> {
+        let config = sshd_output
+            .lines()
+            .filter_map(|line| line.split_once(' '))
+            .collect::<BTreeMap<&str, &str>>();
+
+        let authorized_keys_files: Vec<&str> = config
+            .get("authorizedkeysfile")
+            .unwrap_or(&"none")
+            .split_whitespace()
+            .collect();
+        let authorized_keys_command = config.get("authorizedkeyscommand").unwrap_or(&"none");
+        let authorized_keys_command_user =
+            config.get("authorizedkeyscommanduser").unwrap_or(&"none");
+
+        Ok(Self {
+            authorized_keys_files,
+            authorized_keys_command,
+            authorized_keys_command_user,
+        })
+    }
+}
+
+fn get_keys_from_files(user: &uzers::User, keyfiles: &Vec<&str>) -> Result<String> {
+    let home_dir = user.home_dir();
+    let mut user_authorized_keys = String::new();
+
+    for keyfile in keyfiles {
+        let user_authorized_keys_path = home_dir.join(keyfile);
+
+        if !user_authorized_keys_path.exists() {
+            tracing::debug!(
+                "Skipping authorized key file {} for user {} because it doesn't exist",
+                user_authorized_keys_path.to_string_lossy(),
+                user.name().to_string_lossy()
+            );
+            continue;
+        }
+
+        // Safety: The UID should be valid because we got it from uzers
+        #[allow(unsafe_code)]
+        let user_uid = unsafe { Uid::from_raw(user.uid()) };
```

Comment:
(This is OK for now but I think per previous discussion we actually don't need to change uid for this, we can just use cap-std)